app/Models/DeviceRelatedModel.php

```diff
@@ -46,4 +47,9 @@ public function device()
     {
         return $this->belongsTo('App\Models\Device', 'device_id', 'device_id');
     }
+
+    public function deviceGroup()
+    {
+        return $this->HasMany('App\Models\DeviceGroupDevice', 'device_id', 'device_id');
```

Comment:
You should copy the groups() definition from the device model.  No need to create a model for the pivot table.

Then you just need to make everything consistent.

Reply:
I know it's not needed, but it optimizes the SQL.
Current:
```sql
SELECT *
FROM `eventlog`
WHERE EXISTS
    (SELECT *
     FROM `device_group_device`
     WHERE `eventlog`.`device_id` = `device_group_device`.`device_id`
       AND `device_group_device`.`device_group_id` = '2')
ORDER BY `datetime` DESC
LIMIT 50
OFFSET 0;
```

With the following change:
```php
return $query->whereHas('groups', function ($query) use ($deviceGroup) {
    $query->where('device_groups.id', $deviceGroup);
});
```
The SQL becomes:
```sql
SELECT *
FROM `eventlog`
WHERE EXISTS
    (SELECT *
     FROM `device_groups`
     INNER JOIN `device_group_device` ON `device_groups`.`id` = `device_group_device`.`device_group_id`
     WHERE `eventlog`.`event_id` = `device_group_device`.`device_id`
       AND `device_groups`.`id` = '2')
ORDER BY `datetime` DESC
LIMIT 50
OFFSET 0
```
So this is simply a optimized version.

Reply:
This returns a DeviceGroupDevice model not a DeviceGroup model.